cockatrice/src/chatview.cpp

```diff
@@ -161,72 +161,115 @@ void ChatView::appendMessage(QString message, QString sender, UserLevelFlags use
     }
     cursor.setCharFormat(messageFormat);
     
-    int index = -1, bracketFirstIndex = -1, mentionFirstIndex = -1, urlFirstIndex = -1;
+    int index = -1, bracketFirstIndex = -1, mentionFirstIndex = -1, urlFirstIndex = -1, highlightWordFirstIndex = -1;
     bool mentionEnabled = settingsCache->getChatMention();
     const QRegExp urlStarter = QRegExp("https?://|\\bwww\\.");
     const QRegExp phraseEnder = QRegExp("\\s");
     const QRegExp notALetterOrNumber = QRegExp("[^a-zA-Z0-9]");
+    const QStringList highlightedWords = settingsCache->getHighlightWords();
     
     while (message.size())
     {
-        // search for the first [ or @
         bracketFirstIndex = message.indexOf('[');
         mentionFirstIndex = message.indexOf('@');
         urlFirstIndex = message.indexOf(urlStarter);
+        highlightWordFirstIndex = -1;
+
+        foreach (QString word, message.simplified().split(" "))
```

Comment:
Oh and this wont work with puctuation.

Words:
gambit, cyclops

String:
I like cyclops, he is better than gambit.

Will have no matches.